### PR TITLE
Ft get profile 184638377

### PR DESCRIPTION
**Pull request by dushimsam**
#### What does this PR do?
 
It retrieves the user details of the current loggedin User.

#### Description of  the task to be completed?

Developed the API on which a User can view his/her profile details. 

#### How should this be manually tested?

- checkout into ft-get-profile-184638377 branch and install dependencies 
- Run the login API to get the jsonwebtoken for filling in the authorization, Bearer Token
- Send a GET request to ${Basic URL}/api/users/self

#### Any background context you want to provide?

N/A

#### What are the relevant pivotal tracker stories?
[#184638377](https://www.pivotaltracker.com/n/projects/2633521/stories/184638377)

#### Screenshots (if appropriate)
image

...

### CODE REVIEW

**Review by felixdusengimana — APPROVED** [review]
LGTM

**Review by felixdusengimana — CHANGES_REQUESTED** [review]
Can you include API documentation

**Review by honorine22 — APPROVED** [review]

File: src/routes/userRoute.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Router } from 'express';
+import { authMiddleware } from '../middlewares/authMiddleware';
+import { createUser, getUsers, verifyEmail, welcomeNewUser, loginUser, getMyProfile } from '../controllers/userController';
+const router = Router();
+router.post('/', createUser)
+router.post('/login', loginUser)
+router.use('/verify-email/:token', verifyEmail)
+router.get('/verify-email/:token', welcomeNewUser)
```
Comment (Nkbtemmy) [inline src/routes/userRoute.js:8]: what are the differences???; may be its not yours

File: src/swagger/index.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+dotenv.config();
+
+const host =
+  process.env.NODE_ENV === 'production'
+    ? process.env.BASE_URL.split('https://')[1]
+    : process.env.BASE_URL.split('http://')[1];
+
+const paths = {
+  ...hello
+//   add other defined apis here
```
Comment (Nkbtemmy) [inline src/swagger/index.js:14]: It Seems you didn't document your endpoint !!

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]
Please you have many commits; so squash them into one.

File: src/swagger/index.js (hunk 2 above)
Comment (Nkbtemmy) [reply src/swagger/index.js:14]: Also remember to rebase develop

**Pull request closed without merging**